components/camel-bindy/src/main/java/org/apache/camel/dataformat/bindy/format/factories/ZonedDateTimeFormatFactory.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.dataformat.bindy.format.factories;
+
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Locale;
+
+import org.apache.camel.dataformat.bindy.Format;
+import org.apache.camel.dataformat.bindy.FormattingOptions;
+import org.apache.camel.dataformat.bindy.PatternFormat;
+import org.apache.camel.util.ObjectHelper;
+
+public class ZonedDateTimeFormatFactory extends AbstractFormatFactory {
+
+    {
+        supportedClasses.add(ZonedDateTime.class);
+    }
+
+    @Override
+    public Format<?> build(FormattingOptions formattingOptions) {
+        return new ZonedDateTimePatternFormat(formattingOptions.getPattern(),
+                formattingOptions.getLocale());
+    }
+
+    private static class ZonedDateTimePatternFormat implements PatternFormat<ZonedDateTime> {
+
+        private String pattern;
+        private Locale locale;
+
+        ZonedDateTimePatternFormat(String pattern/*, String timezone*/, Locale locale) {
+            this.pattern = pattern;
+            this.locale = locale;
+        }
+
+        @Override
+        public String format(ZonedDateTime object) throws Exception {
+            ObjectHelper.notNull(this.pattern, "pattern");
+            return this.getDateFormat().format(object);
+        }
+
+        @Override
+        public ZonedDateTime parse(String string) throws Exception {
+
+            ZonedDateTime date;
+            DateTimeFormatter df = this.getDateFormat();
+
+            ObjectHelper.notNull(this.pattern, "pattern");
+            date = ZonedDateTime.parse(string, df);
```

Comment:
I'd change this to initialize `date` directly unless there was another reason perhaps?:
```suggestion
            ZonedDateTime date = ZonedDateTime.parse(string, df);
```